### PR TITLE
[libcommhistory] Give a service name to the object on D-Bus.

### PR DESCRIPTION
**Pull request by dcaliste**
Having a well-known service name attached to the
object when doing a database modification allows
to filter/enable it with Firejail.

@pvuorela, I believe [this issue](https://forum.sailfishos.org/t/removing-entries-from-call-history-programmatically/17909) comes from a Firejail filtering. It's working when the voicecall-ui is launched from the command-line, but not through Sailjail. In the log, there is this issue :
```
B6: <- :1.798 signal com.nokia.commhistory.eventDeleted at /CommHistoryModel
*FILTERED IN*
```

This comes from the fact that the sender is not a well-known service (it's :1.798) and talk and broadcast rules only apply to weel-known service names. The issue is not showing up for addition in the call history for instance, I guess because the event is coming from the voicecall-managerd which owns a well-known service on D-Bus. While, running `commhistory-tool` from the command-line does not own any well-known names.

The solution I'm proposing is to give a well-known name to any process emitting using the com.nokia.commhistory interface.

It also requires to whitelist this new well-known service name in sailjail (see my PR there).

### CODE REVIEW

File: src/dbus_p.h [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #include "messagepart.h"
 #include "group.h"
 
+#define COMM_HISTORY_SERVICE_NAME  QLatin1String("com.nokia.commhistory")
```
Comment (pvuorela) [inline src/dbus_p.h]: Perhaps could be _PREFIX or something as there is no service name with this exact name.

File: src/updatesemitter.cpp [hunk 2]
```diff
@@ -39,6 +40,10 @@ UpdatesEmitter::UpdatesEmitter()
                                                       this)) {
         qCWarning(lcCommHistory) << Q_FUNC_INFO << ": error registering object";
     }
+    if (!QDBusConnection::sessionBus().registerService(QString::fromLatin1("%1.p%2").arg(COMM_HISTORY_SERVICE_NAME).arg(QCoreApplication::applicationPid()))) {
```
Comment (pvuorela) [inline src/updatesemitter.cpp]: Could be nice limiting the line length a bit here. Up for consideration is pid best with potentially different namespaces, but low chance of being a real problem.

**Review by pvuorela — COMMENTED** [review]
Seems mostly good, just some minor comments.

File: src/dbus_p.h (hunk 1 above)
Comment (dcaliste) [reply src/dbus_p.h]: That's right, I've added a _PREFIX to the macro name.

File: src/updatesemitter.cpp (hunk 2 above)
Comment (dcaliste) [reply src/updatesemitter.cpp]: Actually, I noticed that I forgot to unregister the service on the bus. So I'm now saving the service name into a global variable, which has the consequence to reduce the line length !

qtconatcts-sqlite is using a UUID to identify the service. It's a good idea, but it also requires to post-process a bit the UUID string to comform to the D-Bus name specifications. And it gives very long service names, which is not nice when observing the bus and debugging.

**Review by pvuorela — APPROVED** [review]

**Pull request merged**